src/ssl.c

```diff
@@ -14845,10 +14845,12 @@ int wolfSSL_set_compression(WOLFSSL* ssl)
                 ssl->timeoutInfo.timeoutValue.tv_usec = timeout.tv_usec;
                 (toCb)(&ssl->timeoutInfo);
             }
-            /* clean up */
-            FreeTimeoutInfo(&ssl->timeoutInfo, ssl->heap);
             ssl->toInfoOn = 0;
         }
+
+        /* clean up buffers allocated by AddPacketInfo */
+        FreeTimeoutInfo(&ssl->timeoutInfo, ssl->heap);
```

Comment:
Since you move this function, I think it would be good to re-visit `InitTimeoutInfo` and `FreeTimeoutInfo` to meet our coding standards. `InitTimeoutInfo` should just be a `memset` and `FreeTimeoutInfo` should use `NULL` and `{}` around the multi-line `for` loop.

Reply:
Good eye!